### ISSUE TITLE
Settings Sync doesn't show local backup

### ISSUE THREAD

**Issue by borekb** [microsoft/vscode#92641]
Issue Type: <b>Bug</b>

I just signed in to use the new Settings Sync preview but the Local Backup view seems to be empty:

<img width="1189" alt="Screenshot 2020-03-13 at 13 12 35" src="https://user-images.githubusercontent.com/101152/76619967-6f63cf80-652c-11ea-9b74-f3bdb1d1919e.png">

When I open the folder in finder, files are there:

<img width="882" alt="Screenshot 2020-03-13 at 13 13 49" src="https://user-images.githubusercontent.com/101152/76619998-860a2680-652c-11ea-8c1d-534f1ed88358.png">

And they contain my settings:

<img width="1189" alt="Screenshot 2020-03-13 at 13 14 03" src="https://user-images.githubusercontent.com/101152/76620066-a33ef500-652c-11ea-815c-0d8f580efe65.png">

VS Code version: Code - Insiders 1.44.0-insider (efd7548df65de8e775015b67863190e6bd4934fc, 2020-03-13T06:01:51.669Z)
OS version: Darwin x64 19.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz (8 x 2500)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 2, 2|
|Memory (System)|16.00GB (0.83GB free)|
|Process Argv|-psn_0_2359872|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
spellright|ban|3.0.50
markdown-preview-github-styles|bie|0.1.6
vscode-markdownlint|Dav|0.34.0
gitlens|eam|10.2.1
EditorConfig|Edi|0.14.4
vscode-markdown-link-suggestions|Tom|13.0.0
vscodeintellicode|Vis|1.2.6
zoombar-vscode|wra|1.2.1
markdown-all-in-one|yzh|2.7.0

(1 theme extensions excluded)

</details>

**Comment by sandy081** [comment]
@borekb There are no changes done to your local settings. Hence there is nothing to backup. 

`lastSyncSettings.json` is not back up. This is the remote state. Also above view is not an explorer view. This shows history of changes done to your local settings/keybindings/extensions/globalState

**Comment by borekb** [comment]
Ah OK, then maybe the commands have confusing names:

- **Sync: Open Local Backups Folder** – I get that Finder view, with several files in it.
- **Sync: Show Local Backup** – I get an empty view.

If I understand what you say, the first command opens a folder that represents _remote_ state.

**Comment by sandy081** [comment]
First command opens the folder where local back ups are stored. Since there are no backups yet you are not seeing anything either in first or second command.

**Comment by borekb** [comment]
Oh I see. Two possible UX improvements, as I still find the current situation confusing:

1. In the VSCode sidebar, show an explicit label like "No local backups yet" for an empty state.
2. On the file system, maybe use an explicit structure like this:

```
- .../Application Support/Code/User/sync
  - localBackups
  - remoteState
```

The "Sync: Open Local Backups Folder" would open `sync/localBackups`, not `sync`.

**Comment by sandy081** [comment]
> In the VSCode sidebar, show an explicit label like "No local backups yet" for an empty state.

Makes sense. But not sure if it is really needed or not. Let's see if other users also get into this state.

> On the file system, maybe use an explicit structure like this:

Going to folder structure is more advanced and changing the structure for this purpose is not worth. Hence a view is shown representing the local state.

**Comment by borekb** [comment]
I still don't understand what "local backups folder" is. When I invoke that command, it opens `/Users/borekb/Library/Application Support/Code - Insiders/User/sync` – is that it? Or is there supposed to be some subfolder like `localBackups` with even more subfolders per timestamp or something?

BTW, I just changed one setting, restarted VSCode and got this in the Output panel:

```
[2020-03-16 20:40:56.790] [userDataSync] [info] Auto Sync: Started
[2020-03-16 20:40:58.421] [userDataSync] [info] Settings: Updated remote settings
[2020-03-16 20:40:58.436] [userDataSync] [info] Settings: Updated last synchronized settings
[2020-03-16 20:40:58.446] [userDataSync] [info] Keybindings: No changes found during synchronizing keybindings.
[2020-03-16 20:40:58.448] [userDataSync] [info] GlobalState: No changes found during synchronizing ui state.
[2020-03-16 20:40:58.488] [userDataSync] [info] Extensions: No changes found during synchronizing extensions.
[2020-03-16 20:40:58.488] [userDataSync] [info] Sync done. Took 1698ms
```

But still, local backup sidebar is empty at this point and my local backups folder (?) only contains what is supposedly a remote state.

Sorry if this seems insignificant to you, I just want to have confidence in the settings sync mechanism and viewing my previous local settings is important to me, especially since the actual server-side storage is out of my control (might be improved with something like GitHub repo or Gist as a storage: #92357.

**Comment by sandy081** [comment]
> especially since the actual server-side storage is out of my control

BTW We recently added Remote backup view in latest insiders. So your server changes will be seen in this view.

Data is backed up locally whenever it gets changed due to incoming changes. There are remote back ups which stores changes when ever remotes are updated. In your example above, there are outgoing changes and no incoming changes and these changes can be seen in remote back up view. If you have incoming changes, then those changes can be seen in local back up view.

I understand the confusion with local backups and remote back ups and it is with the terminology we used. I understood it while I explained it above. It seems we shall rename them from Local and Remote backups to Incoming & Outgoing changes respectively.
